src/xgov-dapp/src/shared/hooks/useOverflow.ts

```diff
@@ -0,0 +1,25 @@
+import React, {useLayoutEffect} from "react";
+import type {MutableRefObject} from "react";
+
+export const useOverflow = (ref: MutableRefObject<HTMLElement|null>, callback?: (hasOverflow: boolean)=> boolean) => {
```

Comment:
Sadly, this is the "react" way to do native. The `MutatableRefObject` is the reference to the HTMLElement. The `effects` have to handle `null` since the reference is not available on start up. 

It's basically the equivalent of window.onload + document.getElementById but with an initial state of null